### PR TITLE
Create process.rst

### PR DESCRIPTION
**Pull request by orangejenny**
@ctsims Does this seem useful? It's a bit vague. In practice people submitting non-trivial changes are usually working with an internal contact, so my intent is just for this to help orient people.

### CODE REVIEW

File: docs/process.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
+Code Release Process
+^^^^^^^^^^^^^^^^^^^^
+
+This document contains guidance on internal processes that can be relevant when submitting code, in particular to the `commcare-hq <https://github.com/dimagi/commcare-hq/>`_ repository.
```
Comment (ctsims) [inline docs/process.rst]:
```suggestion
This document contains guidance on internal processes that are commonly part of code submission and review in our Open Source repositories.
```

File: docs/process.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+PR Template
+^^^^^^^^^^^
+
+Depending on the repository, your PR may be populated with a structured template. Most of this is oriented around clearly communicating the change being made and accounting for risks in the change.
```
Comment (ctsims) [inline docs/process.rst]:
```suggestion
Depending on the repository, your PR may be populated with a structured template, or in some cases you may have a choice of templates to apply. 

We use templates for a mix of applying guiderails to the submission process, as well as ensuring that the structure of changes is clearly communicated including risks and how they are accounted for.
```

File: docs/process.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+Labels
+^^^^^^
+
+The main Dimagi repos require PRs to be labeled for the sake of internal process. This is enforced by the `label-bot <https://github.com/dimagi/label-bot>`_, which will mark your PR as failing a check if it is unlabeled. A member with write access will label your PR.
```
Comment (ctsims) [inline docs/process.rst]:
```suggestion
Many of our repos require PRs to be labeled for the sake of internal process, with the labels themselves being specific to the contribution process for each repository. 

Label adherence is enforced by the `label-bot <https://github.com/dimagi/label-bot>`_, which will mark your PR as failing a check if it is unlabeled. 

If you do not have core committing access, a member with write access will label your PR based on the code changed, your PR overview, and any other factors that come up during review.
```

**Review by ctsims — APPROVED** [review]
Made some suggestions on wording, mostly since I think it's much better when possible to avoid directly referencing individual codebases for reference (and to push docs into that codebase if unavoidable). 

One thought: It might be the case that this doc primarily wants to communicate about QA, and should be focused there instead of a broader contribution doc. IE: Adding a QA doc to `docs/` and updating the template at https://github.com/dimagi/open-source/blob/master/templates/CONTRIBUTING.rst to reference it.

**Comment by orangejenny** [comment]
@ctsims Cool, thanks for feedback. I'm going to leave it framed as a general process doc, because I do want it to be broader than QA. The label process in particular has been a point of confusion, since if you don't have write access your PR immediately fails that check.

**Pull request merged**